rpc: TestClockOffsetInClientPingRequest failed

rpc.TestClockOffsetInClientPingRequest [failed](https://tanzanite.cluster.engflow.com/invocations/default/93ee62b9-1f4c-4ae2-ba4e-71413ad48610?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvcnBjOnJwY190ZXN0) on release-24.2.1-rc @ [95e6fb913c83be95110e0a250bcc75bbe863c8b1](https://github.com/cockroachdb/cockroach/commits/95e6fb913c83be95110e0a250bcc75bbe863c8b1):


```
=== RUN   TestClockOffsetInClientPingRequest
    context_test.go:228: server listen
    context_test.go:235: client dial
E240905 09:37:02.752509 267 2@rpc/peer.go:598  [rnode=1,raddr=127.0.0.1:39037,class=system,rpc] 11  failed connection attempt (last connected 0s ago): operation "conn heartbeat" timed out after 100ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
    context_test.go:239: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context_test.go:239
        	            				github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context_test.go:188
        	Error:      	Received unexpected error:
        	            	initial connection heartbeat failed: operation "conn heartbeat" timed out after 100ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
        	            	(1) initial connection heartbeat failed
        	            	Wraps: (2) operation "conn heartbeat" timed out after 100ms (given timeout 100ms)
        	            	Wraps: (3) grpc: context deadline exceeded [code 4/DeadlineExceeded]
        	            	Error types: (1) *netutil.InitialHeartbeatFailedError (2) *timeutil.TimeoutError (3) *status.Error
        	Test:       	TestClockOffsetInClientPingRequest
--- FAIL: TestClockOffsetInClientPingRequest (0.12s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClockOffsetInClientPingRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41891

--- a/pkg/rpc/context_test.go
+++ b/pkg/rpc/context_test.go
@@ -235,8 +235,12 @@ func testClockOffsetInPingRequestInternal(t *testing.T, clientOnly bool) {
 	t.Logf("client dial")
 	// Dial: this causes the heartbeats to start.
 	remoteAddr := ln.Addr().String()
-	_, err = rpcCtxClient.GRPCDialNode(remoteAddr, 1, roachpb.Locality{}, SystemClass).Connect(ctx)
-	require.NoError(t, err)
+	testutils.SucceedsSoon(t, func() error {
+		_, err = rpcCtxClient.GRPCDialNode(
+			remoteAddr, 1, roachpb.Locality{}, SystemClass,
+		).Connect(ctx)
+		return err
+	})
 
 	// The first ping establishes the TCP+TLS connection and uses a blocking dialback,
 	// so it's usually pretty noisy in terms of detecting clock offsets. The second

